include/quic/gnutls_crypto.hpp

```diff
@@ -104,10 +129,10 @@ namespace oxen::quic
 
         void set_tls_hook_functions();  // TODO: which and when?
       public:
-        GNUTLSSession(GNUTLSCreds& creds, const ngtcp2_crypto_conn_ref& conn_ref_, bool is_client);
+        GNUTLSSession(GNUTLSCreds& creds, bool is_client);
         ~GNUTLSSession();
 
-        void* get_session() override { return session; };
+        void* get_session() override { return reinterpret_cast<gnutls_session_t>(session); };
```

Comment:
`session` is already a `gnutls_session_t`

Reply:
oops, was checking something debugging a CI build. thanks

Reply:
sorry I clicked resolve on these as we were chatting about them on mumble about it (and some of your later comments), it should be changed